### PR TITLE
SONARJAVA-3666 Do not fail when reporting in text blocks end of lines

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: java-checks-test-sources/src/main/java/checks/regex/InvalidRegexCheck.java [hunk 1]
```diff
@@ -5,14 +5,14 @@
 public class InvalidRegexCheck {
 
   void noncompliant(String str) {
-    Pattern.compile("("); // Noncompliant [[sc=23;ec=24;secondary=8]] {{Fix the syntax error inside this regex.}}
```
Comment (m-g-sonar) [inline java-checks-test-sources/src/main/java/checks/regex/InvalidRegexCheck.java:8]: I'm not at all happy about these changes, which should be OK. I'll handle them.

File: java-checks-test-sources/src/main/files/non-compiling/checks/regex/SingleCharacterAlternationWithTextBlocks.java [hunk 2]
```diff
@@ -8,7 +8,7 @@ void nonCompliant(String str) {
     // Matches [ab\n]
     str.matches("""
       a|b|
-      """); // Noncompliant@-1 [[sc=7;ec=12]]
```
Comment (m-g-sonar) [inline java-checks-test-sources/src/main/files/non-compiling/checks/regex/SingleCharacterAlternationWithTextBlocks.java:11]: should be OK

File: java-frontend/src/test/java/org/sonar/java/regex/JavaAnalyzerRegexSourceTest.java [hunk 3]
```diff
@@ -47,31 +76,35 @@ void test_text_blocks() {
...
-  private void assertTextSpan(int line, int startColumn, int endColumn, AnalyzerMessage.TextSpan span) {
-    assertEquals(line, span.startLine);
-    assertEquals(line, span.endLine);
-    assertEquals(startColumn, span.startCharacter);
-    assertEquals(endColumn, span.endCharacter);
+  private static void assertTextSpan(int line, int startColumn, int endColumn, AnalyzerMessage.TextSpan span) {
+    assertEquals(line, span.startLine, () -> String.format("Expected line to be '%d' but got '%d'", line, span.startLine));
+    assertEquals(line, span.endLine, () -> String.format("Expected line to be '%d' but got '%d'", line, span.endLine));
+    assertEquals(startColumn, span.startCharacter, () -> String.format("Expected start character to be '%d' but got '%d'", startColumn, span.startCharacter));
+    assertEquals(endColumn, span.endCharacter, () -> String.format("Expected end charactger to be '%d' but got '%d'", endColumn, span.endCharacter));
```
Comment (sebastian-hungerecker-sonarsource) [inline java-frontend/src/test/java/org/sonar/java/regex/JavaAnalyzerRegexSourceTest.java:107]: Typo: 'charactger' should be 'character'

**Review by sebastian-hungerecker-sonarsource — APPROVED** [review]
LGTM except for one typo in an assertion message

**Pull request merged**